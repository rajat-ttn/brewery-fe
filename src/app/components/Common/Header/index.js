import React from 'react';
import './style.css';

const Header = ({ toggleSound, isMute }) => {
    const imageName = isMute ? 'alarm-mute.png' : 'alarm.png';
    return (
        <header>
            <img src="../images/beer_logo.png" alt="logo" className="logoStyle"/>
            <h1>The Brewery</h1>
            <span onClick={toggleSound}>
            <img src={`../images/${imageName}`} alt="alarm" className="bellIcon"/>
            </span>
        </header>
    );
}

export default Header;