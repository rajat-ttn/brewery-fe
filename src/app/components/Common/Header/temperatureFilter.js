import React from 'react';
import { PropTypes } from 'prop-types';
import Dropdown from 'react-dropdown';
import 'react-dropdown/style.css';

import { TEMPERATURE_TYPE } from '../../../../constants/textConstants';

const TemperatureFilter = ({ temperatureType, changeTemperatureType }) => {
    let selectedTempLabel = '';
    TEMPERATURE_TYPE.forEach(({ value,label }) => {
        if (temperatureType === value) {
            selectedTempLabel = label;
        }
    });
    return (
        <div className="dropdownSection">
            <Dropdown
                options={TEMPERATURE_TYPE}
                onChange={changeTemperatureType}
                value={selectedTempLabel}
            />
        </div>

    );
};

TemperatureFilter.propTypes = {
    temperatureType: PropTypes.string.isRequired,
    changeTemperatureType: PropTypes.func.isRequired,
};

export default TemperatureFilter;