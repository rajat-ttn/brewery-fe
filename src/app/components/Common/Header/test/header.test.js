import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../index';

describe('Header Component Should render correctly', () => {
    it('renders correctly', () => {
        const HeaderComponent = renderer.create(
            <Header />
        );
        expect(HeaderComponent.toJSON()).toMatchSnapshot();
    });
});
