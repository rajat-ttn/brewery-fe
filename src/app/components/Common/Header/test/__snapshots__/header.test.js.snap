// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should render correctly renders correctly 1`] = `
<header>
  <img
    alt="logo"
    className="logoStyle"
    src="../images/beer_logo.png"
  />
  <h1>
    The Brewery
  </h1>
  <span
    onClick={undefined}
  >
    <img
      alt="alarm"
      className="bellIcon"
      src="../images/beer_logo.png"
    />
  </span>
</header>
`;
